### DISCUSSION TITLE
Reaching data through relationship in related table

### DISCUSSION THREAD

**Discussion by tobehn** [Budibase/budibase#2646] [Help]
I want to set up these Realationships

1 Product has zero or more Assmeblies
1 Product has zero or more Parts
1 Assemblies has 1 or more Parts
1 Part has exactly one Assembly or Product

[![](https://mermaid.ink/img/eyJjb2RlIjoiZXJEaWFncmFtXG4gICAgICAgICAgUFJPRFVDVFMgfW8tLXx7IEFTU0VNQkxJRVMgOiBjb250YWluc1xuICAgICAgICAgIEFTU0VNQkxJRVMgfXwtLXx8IFBBUlRTIDogaGFzXG4gICAgICAgICAgUFJPRFVDVFMgfW8tLXx8UEFSVFMgOiBoYXNcbiAgICAgICAgICBcbiAgICAgICAgICAgICIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2UsImF1dG9TeW5jIjp0cnVlLCJ1cGRhdGVEaWFncmFtIjpmYWxzZX0)](https://mermaid-js.github.io/mermaid-live-editor/edit/##eyJjb2RlIjoiZXJEaWFncmFtXG4gICAgICAgICAgUFJPRFVDVFMgfW8tLXx7IEFTU0VNQkxJRVMgOiBjb250YWluc1xuICAgICAgICAgIEFTU0VNQkxJRVMgfXwtLXx7IFBBUlRTIDogaGFzXG4gICAgICAgICAgUFJPRFVDVFMgfW8tLXx8UEFSVFMgOiBoYXNcbiAgICAgICAgICBcbiAgICAgICAgICAgICIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9)

I think i have done this with the relationships in BB as follows:

One Product row -> Many Assemblies rows
One Product row -> Many Part rows
One Assemblies row -> Many Part rows

I then want to create a formula field containing:

{{Products.0.Number}}-{{ Assemblies.0.Number }}-{{Number}}

this works as long as i have manually defined the product filed and the assembly filed in the part table.
But if i have defined an `product` containing an `assembly` having a few `parts` without setting `product` in all assembly realted parts manually i cannot reach {{Products.0.Number}} through the assembly table.

is there a way to achieve this?

**Comment by mjashanks** [comment]
There is no way to achieve this using the internal tables. I don't see this ability coming in the near future - it's a quite complex behaviour. 

Achievable using SQL though.

**Reply by tobehn** [reply]
This is what i was thinking.
I might have to use an SQL.


Perhaps a bit off topic here:
Is there an adviced way on installing postgreSQL in a docker container locally in the same docker setup when self hosting?

**Reply by mjashanks** [reply]
Depends what you mean by "In the same docker setup". 

I run postgres on docker, locally - I am using Ubuntu. I set this up separately from Budibase, just following the posgres+docker setup instructions. I am then able to connect to it by setting the "host" to `172.17.0.1`, when creating my postgres connection in budibase. If you are using Mac or Windows, `host.docker.internal` should work instead.
